src/esp/assets/ResourceManager.cpp

```diff
@@ -1783,10 +1783,21 @@ bool ResourceManager::loadRenderAssetGeneral(const AssetInfo& info) {
   // Set transformations. Objects that are not part of the hierarchy are
   // ignored, nodes that have no transformation entry retain an identity
   // transformation.
-  for (const Cr::Containers::Pair<unsigned, Mn::Matrix4>& transformation :
+  for (Cr::Containers::Pair<unsigned, Mn::Matrix4>& transformation :
        scene->transformations3DAsArray()) {
     if (Cr::Containers::Optional<esp::assets::MeshTransformNode>& node =
             nodes[transformation.first()]) {
+      if (transformation.second().scaling().product() == 0) {
+        ESP_ERROR() << "0 value in scaling vector for mesh transform. "
+                       "Indicates an issue with source asset. Overriding to "
+                       "small nonzero and continuing.";
+        ESP_ERROR() << transformation.second();
+        for (auto ix = 0; ix < 3; ++ix) {
+          if (transformation.second()[ix][ix] == 0) {
+            transformation.second()[ix][ix] = 1e-7;
```

Comment:
My linear algebra is too rusty to know a better solution than what you have here. Ideally it would add the value to where it should be given a particular rotation, but I have no idea what is the math leading to that.

An alternative option would be checking if the decomposed TRS values are present, via [translationRotationScaling3DFor()](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1SceneData.html#ab401e58557037318e44a8bf1b94f33af), and if they are (which is the case with some but not all glTF files, as we talked about earlier on Slack), then looking just at zero scale components, adjusting them, and composing a matrix out of it again. That way I think it would be robust enough to not even need to print an error message, the error would be printed only if there is just the matrix.